src/pal/src/loader/module.cpp

```diff
@@ -676,7 +676,7 @@ PAL_RegisterModule(
 {
     HINSTANCE hinstance = nullptr;
 
-    int err = PAL_InitializeDLL();
+    int err = PAL_InitializeDLLWithoutSEH();
```

Comment:
Yes, this will break the debugger dlls (mainly mscordaccore) that need the SEH support for address validation.

This might break VS too since they share our PAL.

To get rid of PAL_RegisterModule/UnregisterModule, we need to get rid of any DLLMain handlers required for initialization and explicitly call PAL_InitializeDLL like Jan suggests.  This should be pretty simple for the debugger dlls, but I would have to look into a little more to know for sure.